### PR TITLE
Split currency differences by reconciliations for each transaction

### LINKED ISSUE #343: Improve downloading bank transactions
When you download a split bank transaction with foreign currency invoices, the currency difference should be split in the same manner.

screen shot 2016-12-16 at 3 26 03 pm

Should be bookkept as

screen shot 2016-12-16 at 3 26 57 pm

### PR DESCRIPTION
**Pull request by Baxxx**
Closes #343

### CODE REVIEW

**Comment by Baxxx** [comment]
- [x] Imported bank transaction currency difference lines should have the VAT percent of the accountplan, which is correct (Requested by @hildegunn95)

File: modules/bank/model/bank.class.php [hunk 1]
```diff
@@ -1019,17 +1019,22 @@ public function journal() {
...
-                    $relations_invoices = array();
-                    foreach($relations as $rel) {
-                        $relations_invoices[] = $rel['InvoiceNumber'];
+                    $fb_relations = $fbbank->get_faturabanktransactionrelations($fakturabankID);
+
+                    $relation_groups = array();
+                    $relation_groups_info = array();
+                    foreach ($fb_relations as $rel) {
+                        $relation_groups[$rel['FakturabankReconciliationID']][] = $rel;
+                        $relation_groups_info[$rel['FakturabankReconciliationID']] = array('current_sum' => 0, 'total_sum' => 0);
```
Comment (Baxxx) [inline modules/bank/model/bank.class.php]: Set this in an outside loop

**Review by sls-lnluksa — APPROVED** [review]
Final review with: @stoivo

**Comment by sls-lnluksa** [comment]
⚠️ 
@stoivo Before deploying this, @hildegunn95 should test if everything is correct.
Example: 
Split one bank transaction on multiple reconciliation, and use more reason on each reconciliation.

File: modules/accounting/model/accounting.class.php [hunk 2]
```diff
@@ -1705,6 +1705,59 @@ public function correct_journal_balance($fields, $JournalID, $VoucherType) {
...
+            $fields['voucher_AccountPlanID'] = $result_motkonto_for_currency_difference->AccountPlanID;
+
+            $vat = $this->get_vataccount_object(array('VatID'=>$result_motkonto_for_currency_difference->VatID, 'date' => $fields['voucher_VoucherDate']));
+            $fields['voucher_Vat'] = $vat->Percent;
+            $fields['voucher_VatID'] = $vat->VatID;
+        }
+
+        $fields['voucher_AutoKID']            = '';
+        $fields['voucher_UpdatedByPersonID']  = $_lib['sess']->get_person('PersonID');
+        $fields['voucher_VoucherType'] = $VoucherType;
```
Comment (Baxxx) [inline modules/accounting/model/accounting.class.php]: Align `=`

File: modules/accounting/model/accounting.class.php [hunk 3]
```diff
@@ -1705,6 +1705,59 @@ public function correct_journal_balance($fields, $JournalID, $VoucherType) {
...
+
+            $fields['voucher_AutomaticReason']      = "Automatisk balanse fra valuta forskjell";
+
+            $fields['voucher_AddedByAutoBalance']   = 0;
+            $fields['voucher_Active']               = 1;
+
+            $VoucherID = $_lib['storage']->db_new_hash($fields, 'voucher');
+            $this->set_accountplan_usednow($fields['voucher_AccountPlanID']);
+
+            $this->update_vat_smart(array('VoucherID'=>$VoucherID, 'post'=>$fields, 'comment' => 'Called from: correct_journal_balance'));
```
Comment (Baxxx) [inline modules/accounting/model/accounting.class.php]: This is `add_line_currency_difference`

**Review by stoivo — APPROVED** [review]

**Pull request merged**